tooling/camel-karaf-feature-maven-plugin/src/main/java/org/apache/camel/karaf/feature/maven/EnsureWrapBundleVersionMojo.java

```diff
@@ -0,0 +1,319 @@
+import org.apache.maven.plugin.AbstractMojo;
+import org.apache.maven.plugin.MojoExecutionException;
+import org.apache.maven.plugins.annotations.LifecyclePhase;
+import org.apache.maven.plugins.annotations.Mojo;
+import org.apache.maven.plugins.annotations.Parameter;
+import org.osgi.framework.Version;
+
+@Mojo(name = "ensure-wrap-bundle-version", defaultPhase = LifecyclePhase.PROCESS_RESOURCES)
+public class EnsureWrapBundleVersionMojo extends AbstractMojo {
+
+    public static final String FILE_PROTOCOL = "file:";
+
+    public static final String WRAP_PROTOCOL = "wrap:mvn:";
+    public static final String BUNDLE_VERSION = "Bundle-Version";
+    public static final List<String> HEADERS_AFTER_BUNDLE_VEIRSION = Arrays.asList(
+            //"Bundle-Version",
+            "DynamicImport-Package",
+            "Export-Package",
+            "Export-Service",
+            "Fragment-Host",
+            "Import-Package",
+            "Import-Service",
+            "Provide-Capability",
+            "Require-Bundle",
+            "Require-Capability");
+    
+    private static final String DEFAULT_HEADER = "<?xml version=\"1.0\" encoding=\"UTF-8\" standalone=\"yes\"?>";
+    private static final String LICENCE_HEADER = """
+<?xml version="1.0" encoding="UTF-8"?>
+<!--
+
+    Licensed to the Apache Software Foundation (ASF) under one or more
+    contributor license agreements.  See the NOTICE file distributed with
+    this work for additional information regarding copyright ownership.
+    The ASF licenses this file to You under the Apache License, Version 2.0
+    (the "License"); you may not use this file except in compliance with
+    the License.  You may obtain a copy of the License at
+
+         http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+-->""";
+
+    @Parameter(property = "featuresFilePath", required = true)
+    private String featuresFilePath;
+
+    @Parameter(property = "targetFeature", required = false)
+    private String targetFeature = null;
+
+    public String getFeaturesFilePath() {
+        return featuresFilePath;
+    }
+
+    public void setFeaturesFilePath(String featuresFilePath) {
+        this.featuresFilePath = featuresFilePath;
+    }
+
+    public String getTargetFeature() {
+        return targetFeature;
+    }
+
+    public void setTargetFeature(String targetFeature) {
+        this.targetFeature = targetFeature;
+    }
+
+    @Override
+    public void execute() throws MojoExecutionException {
+        Features featuresData = JaxbUtil.unmarshal(getFeaturesFilePath(), false);
+        List<Feature> features = featuresData.getFeature();
+
+        if (getTargetFeature() != null) {
```

Comment:
The more I think about this feature (the ability to set a target feature), the more I wonder in which case it will be useful in our case. IMHO, it adds extra complexity to the code for limited value. In practice, it should never be used so it could be removed

Reply:
I agree. Now I remember that I did it when I still had the idea of modifying the original feature file and not the one in the target folder.